orchestrator/cosmos_gravity/src/send.rs

```diff
@@ -356,13 +333,46 @@ pub async fn send_to_eth(
         .send_message(
             &[msg],
             Some(MEMO.to_string()),
-            &[fee],
+            &[tx_fee],
             Some(TIMEOUT),
             private_key,
         )
         .await
 }
 
+async fn validate_balance_sufficiency(
+    contact: &Contact,
+    user_address: Address,
+    coins: Vec<Coin>,
+) -> Result<(), CosmosGrpcError> {
+    let mut balances_map: HashMap<String, Uint256> = HashMap::new();
+    let balances = contact.get_balances(user_address).await.unwrap();
+    for balance in balances {
+        balances_map.insert(balance.denom, balance.amount);
+    }
+    for coin in coins {
+        let denom = coin.clone().denom;
+        let denom_amount = balances_map.get(&denom);
+        if denom_amount.is_none() {
+            return Err(CosmosGrpcError::BadInput(format!(
+                "No balance of {} denom",
+                denom,
+            )));
+        }
+        let sub_amount = denom_amount.unwrap().checked_sub(coin.amount).unwrap();
+        if sub_amount < u256!(0) {
```

Comment:
`U256` cannot be negative and the `unwrap` on the `checked_sub` would panic if `denom_amount < coin.amount`. You should do something like
```
if let Some(sub_amount) = denom_amount.unwrap().checked_sub(coin.amount) { /* success */ } else { /* fail */ }
```

Reply:
Updated by adding the additional check before the 'checked_sub'.